jbpm-seldon-prediction-service/src/main/java/org/jbpm/prediction/service/seldon/AbstractSeldonPredictionService.java

```diff
@@ -0,0 +1,167 @@
+        CompositeConfiguration compositeConfiguration = new CompositeConfiguration();
+
+        Configuration javaProperties = new PropertiesConfiguration();
+        Configuration systemProperties = new SystemConfiguration();
+
+        compositeConfiguration.addConfiguration(javaProperties);
+        compositeConfiguration.addConfiguration(systemProperties);
+        try {
+            Configuration config = configs.properties(new File("seldon.properties"));
+            compositeConfiguration.addConfiguration(config);
+        } catch (ConfigurationException e) {
+            logger.debug("Could not find the file 'seldon.properties'. Trying other configuration sources.");
+        }
+
+        final String SELDON_URL = compositeConfiguration.getString(SELDON_URL_KEY);
+
+        if (SELDON_URL == null) {
+            final String errorMessage = "No Seldon endpoint URL specified";
+            logger.error(errorMessage);
+            throw new IllegalArgumentException(errorMessage);
+        }
+
+        logger.debug("Using Seldon endpoint " + SELDON_URL);
+
+        predict = client.target(SELDON_URL).path("predict");
+
+        // set confidence threshold from configuration
+        final String CONFIDENCE_THRESHOLD = compositeConfiguration.getString(CONFIDENCE_THRESHOLD_KEY);
+
+        if (CONFIDENCE_THRESHOLD != null) {
+            try {
+                this.confidenceThreshold = Double.parseDouble(CONFIDENCE_THRESHOLD);
+                logger.info("Setting confidence threshold to " + this.confidenceThreshold);
+            } catch (NumberFormatException e) {
+                logger.error("Invalid confidence threshold in org.jbpm.task.prediction.service.seldon.confidence_threshold");
+            }
+        } else {
+            logger.info("Using default confidence threshold of 1.0");
+        }
+    }
+
+    @Override
+    public abstract String getIdentifier();
+
+    /**
+     * Returns a model prediction given the input data.
+     *
+     * A {@link PredictionRequest} is sent to the Seldon server containing the features built using the concrete
+     * implementation of {@link PredictionRequest#build(List)}. The response is deserialized by
+     * {@link PredictionResponse#parse(String)} and passed to {@link #parsePredictFeatures(PredictionResponse)}
+     * in order to build the {@link PredictionOutcome}.
+     *
+     * @param task Human task data
+     * @param map A map containing the input attribute names as keys and the attribute values as values.
+     * @return A {@link PredictionOutcome} containing the model's prediction for the input data.
+     */
+    @Override
+    public PredictionOutcome predict(Task task, Map<String, Object> map) {
+        final List<List<Double>> features = buildPredictFeatures(task, map);
+        try {
+            final String json = PredictionRequest.build(features);
+            final String stringResponse = predict.request(MediaType.APPLICATION_JSON_TYPE)
```

Comment:
If post request times out, it will throw an exception like
`javax.ws.rs.ProcessingException: RESTEASY004655: Unable to invoke request`
that it's not caught in the catch block.